src/components/chat/chat.ts

```diff
@@ -0,0 +1,335 @@
+    text: string;
+    attachments?: IgcMessageAttachment[];
+  } {
+    return {
+      text: this._chatState?.inputValue,
+      attachments: this._chatState?.inputAttachments,
+    };
+  }
+
+  /**
+   * Controls the chat behavior and appearance through a configuration object.
+   * Use this to toggle UI options, provide suggestions, templates, etc.
+   */
+  @property({ attribute: false })
+  public set options(value: IgcChatOptions) {
+    this._chatState.options = value;
+  }
+
+  public get options(): IgcChatOptions | undefined {
+    return this._chatState.options;
+  }
+
+  /** Returns the default attachments element. */
+  public get defaultAttachments(): TemplateResult {
+    return this._chatInput.defaultAttachmentsArea;
+  }
+
+  /** Returns the default textarea element. */
+  public get defaultTextArea(): TemplateResult {
+    return this._chatInput.defaultTextArea;
+  }
+
+  /** Returns the default file upload button element. */
+  public get defaultFileUploadButton(): TemplateResult {
+    return this._chatInput.defaultFileUploadButton;
+  }
+
+  /** Returns the default send message button element. */
+  public get defaultSendButton(): TemplateResult {
+    return this._chatInput.defaultSendButton;
+  }
+
+  @watch('messages')
+  @watch('draftMessage')
+  @watch('options')
+  protected contextChanged() {
+    this._context.setValue(this._chatState, true);
+  }
+
+  private renderHeader() {
+    return html` <div part="header">
+      <slot name="prefix" part="prefix"></slot>
+      <slot name="title" part="title"
+        >${this._chatState.options?.headerText}</slot
+      >
+      <slot name="actions" part="actions"></slot>
+    </div>`;
+  }
+
+  private renderSuggestions() {
+    return html` <div
+      part="suggestions-container"
+      role="list"
+      aria-label="Suggestions"
+    >
+      <slot name="suggestions-header" part="suggestions-header"></slot>
+      <slot name="suggestions" part="suggestions">
+        ${this._chatState.options?.suggestions?.map(
+          (suggestion) => html`
+            <slot name="suggestion" part="suggestion" role="listitem">
+              <igc-chip
+                @click=${() =>
+                  this._chatState?.handleSuggestionClick(suggestion)}
+              >
+                <span>${suggestion}</span>
+              </igc-chip>
+            </slot>
+          `
+        )}
+      </slot>
+      <slot name="suggestions-actions" part="suggestions-actions"></slot>
+    </div>`;
+  }
+
+  protected override firstUpdated() {
+    this._context.setValue(this._chatState, true);
+  }
+
+  protected override createRenderRoot(): HTMLElement | DocumentFragment {
+    const root = super.createRenderRoot();
+    root.addEventListener('slotchange', () => {
+      this._hasContent =
+        this._chatState?.hasSlotContent('prefix') ||
+        this._chatState?.hasSlotContent('title') ||
+        this._chatState?.hasSlotContent('actions');
+    });
+    return root;
+  }
```

Comment:
We do have a [slot controller](https://github.com/IgniteUI/igniteui-webcomponents/blob/master/src/components/common/controllers/slot.ts) for this kind of stuff.